### ISSUE TITLE
tensorflow jit_compile=True threads leak

### ISSUE THREAD

**Issue by ZhinianQIN** [tensorflow/tensorflow#56752]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.9

### Custom Code

No

### OS Platform and Distribution

Ubuntu 20.04.4 LTS

### Python version

3.8

### Current Behaviour?

```shell
threads leak
```

### Standalone code to reproduce the issue

```shell
def makeModel()
    #Create a normal model...

    #important
    model.compile(optimizer=optimizers.SGD(learning_rate=0.01), loss=losses.MeanAbsoluteError(), jit_compile=True)

    return model


strategy = tf.distribute.MirroredStrategy(
    cross_device_ops=tf.distribute.NcclAllReduce())

with strategy.scope():
    model = makeModel()

model.fit(....
```

### Relevant log output

```shell
cat /proc/{pid of the process}/status


Name:   python3
Umask:  0022
State:  S (sleeping)
Tgid:   134252
Ngid:   0
Pid:    134252
PPid:   69587
TracerPid:      0
Uid:    0       0       0       0
Gid:    0       0       0       0
FDSize: 512
Groups: 0
NStgid: 134252
NSpid:  134252
NSpgid: 134252
NSsid:  69587
VmPeak: 249801300 kB
VmSize: 249801300 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:  37140600 kB
VmRSS:  37140600 kB
RssAnon:        33552768 kB
RssFile:         1614936 kB
RssShmem:        1972896 kB
VmData: 95053716 kB
VmStk:       160 kB
VmExe:      2784 kB
VmLib:   2600056 kB
VmPTE:    194828 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
CoreDumping:    0
THP_enabled:    1
Threads:        29850
SigQ:   0/256322
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000001001000
SigCgt: 0000000180000002
CapInh: 0000000000000000
CapPrm: 0000003fffffffff
CapEff: 0000003fffffffff
CapBnd: 0000003fffffffff
CapAmb: 0000000000000000
NoNewPrivs:     0
Seccomp:        0
Speculation_Store_Bypass:       thread vulnerable
Cpus_allowed:   ffff
Cpus_allowed_list:      0-15
Mems_allowed:   00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000001
Mems_allowed_list:      0
voluntary_ctxt_switches:        19571
nonvoluntary_ctxt_switches:     1302
```
</details>

**Comment by ZhinianQIN** [comment]
Threads leak only in tf.distribute.MirroredStrategy

**Comment by ZhinianQIN** [comment]
`# conding=utf8

from tensorflow.keras import Input, layers, optimizers, losses, metrics
from tensorflow.keras import mixed_precision
import tensorflow as tf


# 设置全局混合精度
policy = mixed_precision.Policy('mixed_float16')
mixed_precision.set_global_policy(policy)


def makeModel():

    input_1 = Input(shape=(1))

    dense1 = layers.Dense(256, activation='leaky_relu')(input_1)
    dense2 = layers.Dense(256, activation='leaky_relu')(dense1)
    dense3 = layers.Dense(256, activation='leaky_relu')(dense2)
    dense4 = layers.Dense(256, activation='leaky_relu')(dense3)

    outputs = layers.Dense(1)(dense4)

    model = tf.keras.Model(input_1, outputs)
    model.summary()

    model.compile(optimizer=optimizers.SGD(learning_rate=0.01),
                  loss=losses.MeanAbsoluteError(), jit_compile=True)

    return model



def _parse_func(serialized):
    return tf.cast(serialized / tf.int64.max, tf.float32), tf.random.uniform([1])

BATCH_SIZE = 24

dataset = tf.data.Dataset.random()
dataset = dataset.take(1024000)
dataset = dataset.map(_parse_func, tf.data.AUTOTUNE, False)
dataset = dataset.batch(BATCH_SIZE, drop_remainder=True)


strategy = tf.distribute.MirroredStrategy(
    cross_device_ops=tf.distribute.HierarchicalCopyAllReduce())

with strategy.scope():
    model = makeModel()

model.fit(dataset, epochs=100)`

Need multiple gpus to reproduce the bug

**Comment by rivershah** [comment]
Yes this issue still persists. Please look at this issue as well: #56423

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
